dds/DCPS/DataSampleHeader.inl

```diff
@@ -161,3 +161,10 @@ OpenDDS::DCPS::DataSampleHeader::test_flag(DataSampleHeaderFlag flag,
   // Test flag bit.
   return base[FLAGS_OFFSET] & mask_flag(flag);
 }
+
+ACE_INLINE bool OpenDDS::DCPS::DataSampleHeader::valid_data() const {
```

Comment:
This function name is a little misleading to me. Something like "is_unregister_or_dispose()" seems more appropriate.

Reply:
It is a little but at the moment, but this function decides what samples are marked as having valid data, which is implementation defined. We just changed it, so I image it could change again. At least that was my thought process behind this function.